iam/api-client/src/main/java/iam/snippets/DeleteServiceAccountKey.java

```diff
@@ -17,66 +17,46 @@
 
 // [START iam_delete_key]
 
-import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
-import com.google.api.client.json.gson.GsonFactory;
-import com.google.api.services.iam.v1.Iam;
-import com.google.api.services.iam.v1.IamScopes;
-import com.google.auth.http.HttpCredentialsAdapter;
-import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.iam.admin.v1.IAMClient;
+import com.google.iam.admin.v1.DeleteServiceAccountKeyRequest;
+import com.google.iam.admin.v1.KeyName;
+
 import java.io.IOException;
-import java.security.GeneralSecurityException;
-import java.util.Collections;
 
 public class DeleteServiceAccountKey {
 
-  // Deletes a service account key.
-  public static void deleteKey(String projectId, String serviceAccountName,
-      String serviceAccountKey) {
-    // String projectId = "my-project-id";
-    // String serviceAccountName = "my-service-account-name";
-    // String serviceAccountKey = "key-name";
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace the variables before running the sample.
+    String projectId = "your-project-id";
+    String serviceAccountName = "my-service-account-name";
+    String serviceAccountKey = "key-name";
 
-    Iam service = null;
-    try {
-      service = initService();
-    } catch (IOException | GeneralSecurityException e) {
-      System.out.println("Unable to initialize service: \n" + e);
-      return;
-    }
+    deleteKey(projectId, serviceAccountName, serviceAccountKey);
+  }
 
+  // Deletes a service account key.
+  public static void deleteKey(String projectId, String serviceAccountName,
+                               String serviceAccountKey) throws IOException {
     // Construct the service account email.
     // You can modify the ".iam.gserviceaccount.com" to match the service account name in which
     // you want to delete the key.
     // See, https://cloud.google.com/iam/docs/creating-managing-service-account-keys?hl=en#deleting
-    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
-    try {
-      String keyToDelete = String.format("projects/-/serviceAccounts/%s/keys/%s",
-          serviceAccountEmail, serviceAccountKey);
+    try (IAMClient iamClient = IAMClient.create()){
+      String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
+
+      String name = KeyName.of(projectId, serviceAccountName, serviceAccountKey)
+              .toString()
+              .replace(serviceAccountName, serviceAccountEmail);
```

Comment:
Is it possible to pass the service account email directly in KeyName instead of replacing it?